django_declarative_apis/machinery/filtering.py

```diff
@@ -67,6 +68,36 @@ def _get_unexpanded_field_value(inst, field_name, field_type):
         return {display_key: getattr(obj, display_key)}
 
 
+def _is_relation(field):
+    return field.is_relation or getattr(field, "is_fake_relation", False)
+
+
+def _is_reverse_relation(field):
+    return isinstance(field, ForeignObjectRel)
+
+
+def _cache_related_instance(inst, field_name, model_cache):
+    try:
+        field_meta = inst._meta.get_field(field_name)
+    except (AttributeError, FieldDoesNotExist):
+        return
+    if not _is_relation(field_meta) or _is_reverse_relation(field_meta):
+        return  # no `attname` in reverse relations
+    val_pk = getattr(inst, field_meta.attname)
```

Comment:
can't we just wrap this in a try catch block and log the error or reraise the error?

Reply:
the problem is that some fields have `attname` but are not relations so we cannot cache them

Reply:
ah ok.